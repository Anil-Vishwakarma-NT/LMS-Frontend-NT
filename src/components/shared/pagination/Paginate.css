.paginate-logo {
    width: 1rem;
}

.paginate-prev-btn{
    display: flex;
    font-size: 0.8rem;
    cursor: pointer;
    gap: 1rem;
}

/* .paginate-prev-btn:hover{
    transform: scale(1.1);
    transition: 0.2s;
} */

.paginate-parent {
    display: flex;
    align-items: center;
    gap: 2rem;
}
.paginate-page-num{
    font-size: 0.65rem;
    color: rgb(154, 154, 154);
}

.paginate-prev-btn.disabled {
  opacity: 0.5;
  /* pointer-events: none; */
  cursor: not-allowed;
}