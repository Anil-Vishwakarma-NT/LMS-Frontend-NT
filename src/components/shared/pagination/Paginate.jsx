import React from "react";
import previous from "../../../assets/previous.png";
import next from "../../../assets/next.png";
import "./Paginate.css";

const Paginate = ({ totalPages, currentPage, onPageChange }) => {
  
  const handleNext = () => {
    if (totalPages > currentPage+1) {
      onPageChange(currentPage + 1);
    }
  };

  const handlePrev = () => {
    if (currentPage > 0) {
      onPageChange(currentPage - 1);
    }
  };

  return (
    <div className="paginate-parent">
      <div className={`paginate-prev-btn ${currentPage === 0 ? "disabled" : ""}`}
        onClick={currentPage === 0 ? null : handlePrev}>
        <img src={previous} alt="previous" className="paginate-logo" />
        <div className="prev-text">Previous</div>
      </div>
        <div className="paginate-page-num">
            Page {currentPage+1} of {totalPages}
        </div>
      <div className={`paginate-prev-btn ${currentPage + 1 === totalPages ? "disabled" : ""}`}
        onClick={currentPage + 1 === totalPages ? null : handleNext}>
        <div className="prev-text">Next</div>
        <img src={next} alt="next" className="paginate-logo" />
      </div>
    </div>
  );
};

export default Paginate;
