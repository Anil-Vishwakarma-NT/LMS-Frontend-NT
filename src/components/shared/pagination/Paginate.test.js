import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Paginate from "./Paginate";

test("renders pagination with correct current and total pages", () => {
  render(<Paginate totalPages={5} currentPage={2} onPageChange={() => {}} />);
  expect(screen.getByText(/Page 3 of 5/i)).toBeInTheDocument();
});

test("calls onPageChange when Next button is clicked", () => {
  const onPageChangeMock = jest.fn();
  render(<Paginate totalPages={5} currentPage={2} onPageChange={onPageChangeMock} />);
  const nextButton = screen.getByText(/Next/i).closest("div");
  
  fireEvent.click(nextButton);
  expect(onPageChangeMock).toHaveBeenCalledWith(3);
});

test("calls onPageChange when Previous button is clicked", () => {
  const onPageChangeMock = jest.fn();
  render(<Paginate totalPages={5} currentPage={2} onPageChange={onPageChangeMock} />);
  const prevButton = screen.getByText(/Previous/i).closest("div");
  
  fireEvent.click(prevButton);
  expect(onPageChangeMock).toHaveBeenCalledWith(1); 
});

test("does not call onPageChange if Next is clicked on the last page", () => {
  const onPageChangeMock = jest.fn();
  render(<Paginate totalPages={5} currentPage={4} onPageChange={onPageChangeMock} />);
  const nextButton = screen.getByText(/Next/i).closest("div");

  fireEvent.click(nextButton);
  expect(onPageChangeMock).not.toHaveBeenCalled(); 
});

test("does not call onPageChange if Previous is clicked on the first page", () => {
  const onPageChangeMock = jest.fn();
  render(<Paginate totalPages={5} currentPage={0} onPageChange={onPageChangeMock} />);
  const prevButton = screen.getByText(/Previous/i).closest("div");

  fireEvent.click(prevButton);
  expect(onPageChangeMock).not.toHaveBeenCalled();
});

test("matches snapshot", () => {
  const { asFragment } = render(<Paginate totalPages={5} currentPage={2} onPageChange={() => {}} />);
  expect(asFragment()).toMatchSnapshot(); 
});
