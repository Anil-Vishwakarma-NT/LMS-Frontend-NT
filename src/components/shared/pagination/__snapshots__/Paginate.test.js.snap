// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="paginate-parent"
  >
    <div
      class="paginate-prev-btn "
    >
      <img
        alt="previous"
        class="paginate-logo"
        src="previous.png"
      />
      <div
        class="prev-text"
      >
        Previous
      </div>
    </div>
    <div
      class="paginate-page-num"
    >
      Page 3 of 5
    </div>
    <div
      class="paginate-prev-btn "
    >
      <div
        class="prev-text"
      >
        Next
      </div>
      <img
        alt="next"
        class="paginate-logo"
        src="next.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
